src/search/ScenarioLikeSearch.ts

```diff
@@ -0,0 +1,56 @@
+import { create, insert, Orama, search } from '@orama/orama'
+
+import { TypedIndex } from './types.js'
+
+const schema = {
+  name: 'string',
+  description: 'string',
+} as const
+
+export interface ScenarioLike {
+  id: string
+  name: string
+  description: string
+}
+
+/**
+ * Can be used for Backgrounds, Scenarios and Rules, searching against the
+ * name and description
+ */
+class ScenarioLikeSearch<T extends ScenarioLike> implements TypedIndex<T> {
```

Comment:
Uses duck typing to avoid the need for specific classes for Scenario, Rule or Background